### DISCUSSION TITLE
How to get rotate angle

### DISCUSSION THREAD

**Discussion by zaferguler87** [mindee/doctr#961] [Q&A] [answered]
I have a rotated image. With the "db_resnet50_rotation" model, I can successfully find the text in the image. How can I find how many degrees the image or text box is rotated?

**Comment by frgfm** [comment]
Hello @zaferguler87 :wave:

Interesting question! Let's try to give you all the relevant information:
- the model itself (db_resnet50_rotation) is trained on samples that have both rotated and straight text boxes
- that model will produce a raw segmentation map for those
- depending on the `assume_straight_pages` & `export_as_straight_boxes` attribute values, its postprocessor will compute either a straight bounding box or a polygon that fit each word. The polygon are oriented!

That's it for the detection part, if you use the full `ocr_predictor` depending on the attributes, the exported Page will have all this information :+1: 

I hope this helps! If you want us to help you, could you share a minimal runnable code snippet to show how you use your model currently?

**Comment by zaferguler87** [comment, marked as answer]
Thanks for your inspiring comments.

You can find the basic code block below.

```
from doctr.io import DocumentFile
from doctr.models import ocr_predictor

doc = DocumentFile.from_images("bill/11.jpeg")
predictor = ocr_predictor(pretrained=True, assume_straight_pages=False, det_arch='db_resnet50_rotation', export_as_straight_boxes=False, preserve_aspect_ratio=True)
result = predictor(doc)

result.show(doc)

json_export = result.export()
print(json_export)
```
After changing the `export_as_straight_boxes `variable to `False `I noticed that in my case each word is represented as a 4-point polygon as seen below.

![image](https://user-images.githubusercontent.com/59082455/175773198-f682ed70-5179-42c3-a8dd-4010c77f35cc.png)
![image](https://user-images.githubusercontent.com/59082455/175773576-be8141cc-2e12-4007-8270-0e4d92ab0d07.png)


In addition, the points for each word come in clockwise order. Therefore, with a little analytical geometry, the rotation angle can be obtained for each box. Considering the mean or median values in all box angles, the rotate angle of the image can be computed approximately.

**Reply by frgfm** [reply]
Yes exactly :) 
Does it solve your problem then?

**Reply by zaferguler87** [reply]
Yes the problem is solved. Thank you for your help
